arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -69,16 +73,499 @@
+        id = arangodb::RocksDBValue::documentId(value);
+        break;
+      }
+      id = arangodb::RocksDBKey::indexDocumentId(key);
+      break;
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_FULLTEXT_INDEX:
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_GEO1_INDEX:
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_GEO2_INDEX:
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_GEO_INDEX:
+      id = arangodb::RocksDBKey::indexDocumentId(key);
+      break;
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_IRESEARCH_LINK:
+    case arangodb::RocksDBIndex::TRI_IDX_TYPE_UNKNOWN:
+    default:
+      break;
+  }
+
+  return id;
+}
+
+TRI_voc_rid_t getRevisionFromOldDocumentId(rocksdb::DB& db,
+                                           arangodb::RocksDBMetaCollection& rcoll,
+                                           arangodb::LocalDocumentId oldId) {
+  arangodb::RocksDBKey key;
+  key.constructDocument(rcoll.objectId(), oldId);
+  rocksdb::PinnableSlice ps;
+  rocksdb::Status s =
+      db.Get(rocksdb::ReadOptions(), arangodb::RocksDBColumnFamily::documents(),
+             key.string(), &ps);
+  if (!s.ok()) {
+    return TRI_voc_rid_t{0};
+  }
+
+  arangodb::velocypack::Slice doc = arangodb::RocksDBValue::data(ps);
+  return arangodb::transaction::helpers::extractRevFromDocument(doc);
+}
+
+void rewritePrimaryIndexEntry(arangodb::RocksDBIndex& ridx, arangodb::RocksDBKey& key,
+                              arangodb::RocksDBValue& buffer, rocksdb::Slice& value,
+                              rocksdb::Slice oldKey, rocksdb::Slice oldValue,
+                              arangodb::LocalDocumentId& newId) {
+  arangodb::velocypack::StringRef docKey = arangodb::RocksDBKey::primaryKey(oldKey);
+  TRI_voc_rid_t revision = arangodb::RocksDBValue::revisionId(oldValue);
+  key.constructPrimaryIndexValue(ridx.tempObjectId(), docKey);
+  buffer = arangodb::RocksDBValue::PrimaryIndexValue(newId, revision);
+  value = {buffer.string().data(), buffer.string().size()};
+}
+
+void rewriteEdgeIndexEntry(arangodb::RocksDBIndex& ridx, arangodb::RocksDBKey& key,
+                           rocksdb::Slice oldKey, arangodb::LocalDocumentId& newId) {
+  arangodb::velocypack::StringRef vertexId = arangodb::RocksDBKey::vertexId(oldKey);
+  key.constructEdgeIndexValue(ridx.tempObjectId(), vertexId, newId);
+}
+
+void rewriteVPackIndexEntry(arangodb::RocksDBIndex& ridx, arangodb::RocksDBKey& key,
+                            arangodb::RocksDBValue& buffer, rocksdb::Slice& value,
+                            rocksdb::Slice oldKey, arangodb::LocalDocumentId& newId) {
+  arangodb::velocypack::Slice indexedValues = arangodb::RocksDBKey::indexedVPack(oldKey);
+  if (ridx.unique()) {
+    key.constructUniqueVPackIndexValue(ridx.tempObjectId(), indexedValues);
+    buffer = arangodb::RocksDBValue::UniqueVPackIndexValue(newId);
+    value = {buffer.string().data(), buffer.string().size()};
+    return;
+  }
+  key.constructVPackIndexValue(ridx.tempObjectId(), indexedValues, newId);
```

Comment:
if we get here, `value` is left unmodified. Is that intentional?

Reply:
Yes, The non-unique index entry stores the revision in the key rather than the value.